### ISSUE TITLE
Not compatible with ScreenshotExtension

### ISSUE THREAD

**Issue by mathieucarbou** [concordion/concordion-markdown-extension#2]
PNG files are renamed to .html file because of the FileTargetWithSuffix class.

One way to fix that is to use this `Target` to only rename `md` to `html` and use it in the MardownExtensions:

```
private Target target = new FileTargetSuffixReplacer("md", "html");
```

```
public class FileTargetSuffixReplacer implements Target {

  private static final String PROPERTY_OUTPUT_DIR = "concordion.output.dir";
  private String previousSuffix;
  private String suffix;
  private FileTarget target;

  public FileTargetSuffixReplacer(String previousSuffix, String suffix) {
    this.suffix = suffix;
    this.previousSuffix = previousSuffix;
    target = new FileTarget(getBaseOutputDir());
  }

  private File getBaseOutputDir() {
    String outputPath = System.getProperty(PROPERTY_OUTPUT_DIR);
    if (outputPath == null) {
      return new File(System.getProperty("java.io.tmpdir"), "concordion");
    }
    return new File(outputPath);
  }


  @Override
  public void write(Resource resource, String s) throws IOException {
    target.write(rename(resource), s);
  }

  @Override
  public void copyTo(Resource resource, InputStream inputStream) throws IOException {
    target.copyTo(rename(resource), inputStream);
  }

  @Override
  public OutputStream getOutputStream(Resource resource) throws IOException {
    return target.getOutputStream(rename(resource));
  }

  @Override
  public void delete(Resource resource) throws IOException {
    target.delete(rename(resource));
  }

  @Override
  public boolean exists(Resource resource) {
    return target.exists(rename(resource));
  }

  @Override
  public String resolvedPathFor(Resource resource) {
    return target.resolvedPathFor(rename(resource));
  }

  private Resource rename(Resource resource) {
    if(resource.getName().endsWith("." + previousSuffix)) {
      return new Resource(resource.getPath().replaceFirst("\\..*$", "\\." + suffix));  
    } else {
      return resource;
    }
  }
}
```

**Comment by nigelcharman** [comment]
Thanks. We just found this with the ExcelExtension last week too. Will apply a fix in the next few days.

If you're trying out this extension, I'd be keen on any feedback. We're trialling different syntaxes on some different branches. The alternateSyntax branch is the most complete at this stage. Examples will be the next tag to add. The syntax3 branch looks like it won't work.

**Comment by nigelcharman** [comment]
Fixed by 4f52730
